codalab/lib/bundle_cli.py

```diff
@@ -1215,30 +1260,36 @@ def do_rm_command(self, args):
     )
     def do_search_command(self, args):
         client, worksheet_uuid = self.parse_client_worksheet_uuid(args.worksheet_spec)
-        bundle_uuids = client.search_bundle_uuids(worksheet_uuid, args.keywords)
-        if not isinstance(bundle_uuids, list):  # Direct result
-            print >>self.stdout, bundle_uuids
-            return
+        rest_client, _ = self.parse_client_worksheet_uuid(args.worksheet_spec, use_rest=True)
 
-        # print >>self.stdout, out bundles
-        bundle_infos = client.get_bundle_infos(bundle_uuids)
-        bundle_info_list = [bundle_infos[uuid] for uuid in bundle_uuids if uuid in bundle_infos]
-        reference_map = self.create_reference_map('bundle', bundle_info_list)
-        if args.uuid_only:
-            bundle_info_list = [{'uuid': uuid} for uuid in bundle_uuids]
+        bundles = rest_client.fetch('bundles', params={
+            'worksheet': worksheet_uuid,
+            'keywords': args.keywords,
+            'include': ['users', 'bundle-permissions', 'bundles'],
+        })
 
-        if len(bundle_info_list) > 0:
-            self.print_bundle_info_list(bundle_info_list, uuid_only=args.uuid_only, print_ref=False)
+        # Print direct numeric result
+        if 'meta' in bundles:
```

Comment:
It seems like `bundles` can either be a list of a dict, which is not ideal.  I know this was how it was before, but given that we're cleaning things up and exposing the REST API publicly, let's figure out if there's a good convention here.